vhdbuilder/packer/vhd-image-builder-base.json

```diff
@@ -87,6 +87,11 @@
         "sudo chown -R $USER /opt/certs"
       ]
     },
+    {
+      "type": "file",
+      "source": "pkg/parser/bin/parser",
```

Comment:
so if I were you I'd probably have it such that we build the binary during the VHD build itself - that way we don't have to host binaries within the repository, and we can also maintain the validity checks it seems you want via the VHD content tests

Reply:
I kinda sorta did something similar for the prefetch script generation: https://github.com/Azure/AgentBaker/pull/3971

Reply:
Anuj said this might be difficult because we need access to goproxy to build the go binary. Is there an easy way to add goproxy here?

Reply:
I wouldn't think we'd need to use the goproxy since the parser doesn't need internal deps as far as I can see - though if we did need it then we could simply utilize the devinfra PAT/goproxy static token at build-time, similarly to how I added the same for the auto-vendor pipeline the other day

Reply:
makes me wonder though, why are we building the parser as a separate main package with the agentbaker module? 

https://github.com/Azure/AgentBaker/blob/master/pkg/parser/cmd/main.go


I'd probably prefer for the parser to be in its own module which specifies agentbaker/nbcontract as dependencies

Reply:
we do have an internal dependency though, we import the contract in parser, so it needs to pull it.

Reply:
even if it were its own module, we'd need to import the whole thing through packer and onto the VM to compile it during build time, which I am not sure if its the best thing to do.

or are you suggestion something else here?

Reply:
I think I understand what Cameron means, however, we'd still need the PAT

Reply:
the contract is hosted as an agentbaker package through right? agentbaker is OSS so shouldn't need the goproxy afaik

if it were an RP module/pkg then it would require the proxy is my understanding

Reply:
agentbaker is OSS but it still comes under `github.com/Azure/agentbaker`, why I am also certain it needs goproxy is because the compilation of the go binary failed when I hadn't updated my PAT, and it tried to import the package :)

Reply:
seems like no goproxy works, im assuming what happened in your case was you still had you GOPROXY vars setup using an invalid PAT, in which case go just blindly tried to use the proxy and thus it failed